src/server.rs

```diff
@@ -49,6 +49,9 @@ impl<D: Sync + Send + 'static> Server<D> {
             server.keep_alive(timeout);
         }
 
-        server.handle(arc)
+        match thread_count {
```

Comment:
I think there is an advantage to being able to use whatever defaults hyper provides. I presume they have tuned it for the common use cases, and we are providing overrides for less common ones. I think we should just note in the docs that 'None' will use hyper's default. If they improve their default calculations, we can benefit without needing to do anything.